arroyo/processing/strategies/dead_letter_queue/policies/count.py

```diff
@@ -0,0 +1,60 @@
+from collections import deque
+from time import time
+from typing import NamedTuple, Sequence, Tuple
+
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessage,
+)
+from arroyo.types import TPayload
+from arroyo.utils.metrics import get_metrics
+
+
+class _Bucket(NamedTuple):
+    """
+    1-second time bucket to count number of hits in this second.
+    """
+
+    timestamp: int
+    hits: int
+
+
+class CountInvalidMessagePolicy(DeadLetterQueuePolicy[TPayload]):
+    """
+    Ignore invalid messages up to a certain limit per time unit window in seconds.
+    This window is 1 minute by default. The exception associated with the invalid
+    message is raised for all incoming invalid messages which go past this limit.
+
+    A saved state in the form `[(<timestamp: int>, <hits: int>), ...]` can be passed
+    on init to load the previously saved state. This state should be aggregated to
+    1 second buckets.
+    """
+
+    def __init__(
+        self, limit: int, seconds: int = 60, load_state: Sequence[Tuple[int, int]] = []
```

Comment:
Do we still need load_state? I thought we're having a separate policy that does loading from Redis now?

Reply:
I'm thinking that separate policy should inherit this policy, so for it to load a state when initialized I made this one loadable.

So it could do something like:
```python3
def __init__(...):
    super().__init__(limit, seconds, load_state)
    self._init_redis_stuff()
```

Otherwise there would need to be some method in the base class to load a state or something. The other option is to just not inherit from this class at all.

Reply:
I see. One option would be to make `self.hits` not private. Or you can just keep it this way, but if you do please make it:
```
load_state: Optional[Sequence[Tuple[int, int]]] = None
```

due to https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments